ReactAndroid/src/main/java/com/facebook/react/uimanager/ViewGroupManager.java

```diff
@@ -61,54 +62,10 @@ public void addViews(T parent, List<View> views) {
 
   public static void setViewZIndex(View view, int zIndex) {
     mZIndexHash.put(view, zIndex);
-    // zIndex prop gets set BEFORE the view is added, so parent may be null.
-    ViewGroup parent = (ViewGroup) view.getParent();
-    if (parent != null) {
-      reorderChildrenByZIndex(parent);
-    }
   }
 
-  public static void reorderChildrenByZIndex(ViewGroup view) {
-    // Optimization: loop through the zIndexHash to test if there are any non-zero zIndexes
-    // If there aren't any, we can just return out
-    Collection<Integer> zIndexes = mZIndexHash.values();
-    boolean containsZIndexedElement = false;
-    for (Integer zIndex : zIndexes) {
-      if (zIndex != 0) {
-        containsZIndexedElement = true;
-        break;
-      }
-    }
-    if (!containsZIndexedElement) {
-      return;
-    }
-
-    // Add all children to a sortable ArrayList
-    ArrayList<View> viewsToSort = new ArrayList<>();
-    for (int i = 0; i < view.getChildCount(); i++) {
-      viewsToSort.add(view.getChildAt(i));
-    }
-    // Sort the views by zIndex
-    Collections.sort(viewsToSort, new Comparator<View>() {
-      @Override
-      public int compare(View view1, View view2) {
-        Integer view1ZIndex = mZIndexHash.get(view1);
-        if (view1ZIndex == null) {
-          view1ZIndex = 0;
-        }
-
-        Integer view2ZIndex = mZIndexHash.get(view2);
-        if (view2ZIndex == null) {
-          view2ZIndex = 0;
-        }
-        return view1ZIndex - view2ZIndex;
-      }
-    });
-    // Call .bringToFront on the sorted list of views
-    for (int i = 0; i < viewsToSort.size(); i++) {
-      viewsToSort.get(i).bringToFront();
-    }
-    view.invalidate();
+  public static @Nullable Integer getViewZIndex(View view) {
+    return mZIndexHash.get(view);
```

Comment:
Right now I'm using null as a way to know that the view doesn't have z-index set so we can avoid using custom view drawing when no child has z-index set.